include/http2.hrl

```diff
@@ -85,8 +85,8 @@
 
 -type stream_id() :: non_neg_integer().
 -record(frame_header, {
-    length      :: non_neg_integer(),
-    type        :: frame_type(),
+    length      :: non_neg_integer() | undefined,
```

Comment:
There's no reason to prevent anyone from building this record in whatever order they choose. If they decide to send it without a length, they deserve the protocol error!